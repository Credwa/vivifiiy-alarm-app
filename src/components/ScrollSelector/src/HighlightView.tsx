import React, { FC } from 'react';
import { View } from 'react-native';

interface IProps {
  wrapperHeight: number;
  itemHeight: number;
  highlightWidth: number | string;
  highlightColor: string;
  highlightBorderWidth: number;
}

const HighlightView: FC<IProps> = ({
  children,
  wrapperHeight,
  itemHeight,
  highlightWidth,
  highlightColor,
  highlightBorderWidth
}) => (
  <View
    style={{
      position: 'absolute',
      top: (wrapperHeight - itemHeight) / 2,
      height: itemHeight,
      width: highlightWidth,
      borderTopColor: highlightColor,
      borderBottomColor: highlightColor,
      borderTopWidth: highlightBorderWidth,
      borderBottomWidth: highlightBorderWidth
    }}
  >
    {children}
  </View>
);

export default HighlightView;
