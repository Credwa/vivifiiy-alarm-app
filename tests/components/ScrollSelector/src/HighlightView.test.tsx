import React from 'react';
import renderer from 'react-test-renderer';
import HighlightView from '@/components/ScrollSelector/src/HighlightView';

describe('ScrollSelector/src/ <HighlightView />', () => {
  it(`renders correctly`, () => {
    const tree = renderer
      .create(
        <HighlightView
          highlightColor={'#333'}
          highlightWidth={'100%'}
          wrapperHeight={180}
          itemHeight={60}
          highlightBorderWidth={2}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
