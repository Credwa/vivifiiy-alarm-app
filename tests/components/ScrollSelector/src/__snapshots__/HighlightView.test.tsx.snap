// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollSelector/src/ <HighlightView /> renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#333",
      "borderBottomWidth": 2,
      "borderTopColor": "#333",
      "borderTopWidth": 2,
      "height": 60,
      "position": "absolute",
      "top": 60,
      "width": "100%",
    }
  }
/>
`;
